src/rosdistro/dependency_walker.py

```diff
@@ -48,7 +49,9 @@ def _get_package(self, pkg_name):
             pkg_xml = self._distribution_instance.get_release_package_xml(pkg_name)
             try:
                 pkg = parse_package_string(pkg_xml)
-            except InvalidPackage as e:
+                if self._evaluate_condition_context:
+                    pkg.evaluate_conditions(self._evaluate_condition_context)
```

Comment:
Please move the lines 52 and 53 into an `else` block after `except` since they can't raise an `InvalidPackage` exception. In that `else` you can use another `try` / `except` for `ValueError` but raising `InvalidPackage` doesn't seem to make sense in that case.